### ISSUE TITLE
Initialization values must have unique scores, but shouldn't

### ISSUE THREAD

**Issue by guillochon** [fmfn/BayesianOptimization#25]
Currently, the `initialize` method requires a dictionary of values and positions to be passed where the values are used as keys. This means that any two positions that share the same value cannot be passed to the routine, as the keys will clash. Internally it appears there is nothing that prevents this change from happening as the positions/values are eventually stored as arrays anyway. Ideally, initialize should accept an array of dictionaries instead, such that positions that share the same value do not clash.

**Comment by fmfn** [comment]
You are absolutely right. Having it accept a list of dict is a very simple change, I'll do it soon, but feel free to submit a PR if you like.
